### PR TITLE
refactor(component): extract "Contribution CTA" to reusable component

### LINKED ISSUE #69: Extract "Contribution CTA" Component to a Reusable Component
We would like to enhance our codebase by extracting the "Contribution CTA" section currently used in the `/editor/index.astro` and `/browse/[alpha]/[page].astro` files into a standalone reusable component. This will improve code maintainability and reusability.

#### Current Implementation
The "Contribution CTA" section is currently duplicated in both `/editor/index.astro` and `/browse/[alpha]/[page].astro`. Below is the code snippet from the `/editor/index.astro` file:

```html
<!-- Contribution CTA -->
<section class="rounded-lg border shadow-sm">
  <div class="space-y-1.5 p-3 md:p-6 flex flex-col lg:flex-row lg:items-center justify-between gap-4">
    <div class="space-y-1">
      <h3 class="font-semibold tracking-tight text-2xl">Contribute Words</h3>
      <p class="text-sm">
        You can contribute new words to the jargons.dev dictionary.
      </p>
    </div>
    <a
      class="flex items-center w-fit justify-center no-underline px-4 py-2 rounded-md border border-gray-200 bg-white hover:bg-gray-100 text-sm shadow-sm hover:shadow transition-colors"
      href="/editor/new"
    >
      <span>Start Now</span>
      <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width={1.5} stroke="currentColor" class="w-4 h-4">
        <path stroke-linecap="round" stroke-linejoin="round" d="m4.5 19.5 15-15m0 0H8.25m11.25 0v11.25" />
      </svg> 
    </a>
  </div>
</section>
```

#### Task
1. **Create a New Component:**
   - Create a new file named `contribution-cta.astro` in the components directory at `src/components`.
   - Move the "Contribution CTA" code into this new file.

2. **Update `/editor/index.astro` and `/browse/[alpha]/[page].astro`:**
   - Replace the existing "Contribution CTA" section in both files with the newly created `contribution-cta.astro` component.

#### Implementation Steps
1. Create a new file `contribution-cta.astro` with the following content:
   ```html
   <section class="rounded-lg border shadow-sm">
     <div class="space-y-1.5 p-3 md:p-6 flex flex-col lg:flex-row lg:items-center justify-between gap-4">
       <div class="space-y-1">
         <h3 class="font-semibold tracking-tight text-2xl">Contribute Words</h3>
         <p class="text-sm">
           You can contribute new words to the jargons.dev dictionary.
         </p>
       </div>
       <a
         class="flex items-center w-fit justify-center no-underline px-4 py-2 rounded-md border border-gray-200 bg-white hover:bg-gray-100 text-sm shadow-sm hover:shadow transition-colors"
         href="/editor/new"
       >
         <span>Start Now</span>
         <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width={1.5} stroke="currentColor" class="w-4 h-4">
           <path stroke-linecap="round" stroke-linejoin="round" d="m4.5 19.5 15-15m0 0H8.25m11.25 0v11.25" />
         </svg> 
       </a>
     </div>
   </section>
   ```

2. In the `/editor/index.astro` file, replace the existing "Contribution CTA" section with:
   ```html
   <ContributeCTA />
   ```

3. In the `/browse/[alpha]/[page].astro` file, replace the existing "Contribution CTA" section with:
   ```html
   <ContributeCTA />
   ```

#### Acceptance Criteria
- The `contribution-cta.astro` component is created and contains the extracted code.
- The `/editor/index.astro` and `/browse/[alpha]/[page].astro` files are updated to use the new `contribution-cta.astro` component.
- The application functions as expected with the new component in place.

#### Additional Information
- Feel free to ask for more details or clarifications if needed. This enhancement aims to improve the maintainability and reusability of our code.

### PR DESCRIPTION
**Pull request by Bridgetamana**
### Description

I extracted the "Contribution CTA" section from /editor/index.astro and /browse/[alpha]/[page].astro files into a standalone reusable component.

### Related Issue

fixes #69 

### Screenshots/Screencasts

screenshot (2)
screenshot (3)
screenshot (4)
screenshot (5)
screenshot

### CODE REVIEW

File: src/pages/browse/[alpha]/[page].astro [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { ALPHABETS } from "../../../../constants.js";
 import BaseLayout from "../../../layouts/base.astro";
 import Navbar from "../../../components/navbar.astro";
 import Search from "../../../components/islands/search";
+import ContributeCTA from "../../../components/contribution-cta.astro";
```
Comment (babblebey) [inline src/pages/browse/[alpha]/[page].astro:7]:
```suggestion
import ContributionCTA from "../../../components/contribution-cta.astro";
```

File: src/pages/browse/[alpha]/[page].astro [hunk 2]
```diff
@@ -55,25 +56,7 @@ export async function getStaticPaths({ paginate }) {
...
-                href="/editor/new"
-              >
-                <span>Start Now</span>
-                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width={1.5} stroke="currentColor" class="w-4 h-4">
-                  <path stroke-linecap="round" stroke-linejoin="round" d="m4.5 19.5 15-15m0 0H8.25m11.25 0v11.25" />
-                </svg> 
-              </a>
-            </div>
-          </section>
+          <ContributeCTA />
```
Comment (babblebey) [inline src/pages/browse/[alpha]/[page].astro:59]:
```suggestion
          <ContributionCTA />
```

File: src/pages/editor/index.astro [hunk 3]
```diff
@@ -6,6 +6,7 @@ import Navbar from "../../components/navbar.astro";
 import { buildStatsUrl } from "../../lib/utils/index.js";
 import { PROJECT_REPO_DETAILS } from "../../../constants.js";
 import doContributionStats from "../../lib/actions/do-contribution-stats.js";
+import ContributeCTA from "../../components/contribution-cta.astro";
```
Comment (babblebey) [inline src/pages/editor/index.astro:9]:
```suggestion
import ContributionCTA from "../../components/contribution-cta.astro";
```

File: src/pages/editor/index.astro [hunk 4]
```diff
@@ -52,25 +53,7 @@ const totalWords = {
...
-          href="/editor/new"
-        >
-          <span>Start Now</span>
-          <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width={1.5} stroke="currentColor" class="w-4 h-4">
-            <path stroke-linecap="round" stroke-linejoin="round" d="m4.5 19.5 15-15m0 0H8.25m11.25 0v11.25" />
-          </svg> 
-        </a>
-      </div>
-    </section>
+    <ContributeCTA />
```
Comment (babblebey) [inline src/pages/editor/index.astro:56]:
```suggestion
    <ContributionCTA />
```

```suggestion
    <ContributionCTA />
```

**Review by babblebey — COMMENTED** [review]
Hey @Bridgetamana, 

Another great stuff right here, this is looking great. 🤌🏾

I found some tiny changes you should make in the component import name `ContributeCTA`, and Yooooo I realize that was my fault 🫣 as it's what I've stated in the steps in the related issue.

You can fix that by committing the suggestion, then we're prime to merge this for sure 😁

**Comment by Bridgetamana** [comment]
Done

File: src/components/contribution-cta.astro [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+          class="flex items-center w-fit justify-center no-underline px-4 py-2 rounded-md border border-gray-200 bg-white hover:bg-gray-100 text-sm shadow-sm hover:shadow transition-colors"
+          href="/editor/new"
+        >
+          <span>Start Now</span>
+          <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width={1.5} stroke="currentColor" class="w-4 h-4">
+            <path stroke-linecap="round" stroke-linejoin="round" d="m4.5 19.5 15-15m0 0H8.25m11.25 0v11.25" />
+          </svg> 
+        </a>
+      </div>
+    </section>
```
Comment (babblebey) [inline src/components/contribution-cta.astro:19]:
```suggestion
<section class="rounded-lg border shadow-sm">
  <div class="space-y-1.5 p-3 md:p-6 flex flex-col lg:flex-row lg:items-center justify-between gap-4">
    <div class="space-y-1">
      <h3 class="font-semibold tracking-tight text-2xl">Contribute Words</h3>
      <p class="text-sm">
        You can contribute new words to the jargons.dev dictionary.
      </p>
    </div>
    <a
      class="flex items-center w-fit justify-center no-underline px-4 py-2 rounded-md border border-gray-200 bg-white hover:bg-gray-100 text-sm shadow-sm hover:shadow transition-colors"
      href="/editor/new"
    >
      <span>Start Now</span>
      <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width={1.5} stroke="currentColor" class="w-4 h-4">
        <path stroke-linecap="round" stroke-linejoin="round" d="m4.5 19.5 15-15m0 0H8.25m11.25 0v11.25" />
      </svg> 
    </a>
  </div>
</section>
```

**Review by babblebey — APPROVED** [review]
Looks good @Bridgetamana 😁 

PS: I just seen your screenshot yea... Great, but the screenshot suited for this PR would've been that of the component on the web page as seen below 😉, not a screenshot of the changes in VSCode - that is already available here in the "File Changes" tab on GitHub

image

**Comment by Bridgetamana** [comment]
Oh okay, noted.
Thank you so much for the feedback, really learning a lot from you

**Pull request merged**